drivers/flash/flash_stm32g4x.c

```diff
@@ -28,17 +31,23 @@ bool flash_stm32_valid_range(const struct device *dev, off_t offset,
 			     uint32_t len,
 			     bool write)
 {
+
+#if defined(FLASH_OPTR_DBANK) && (CONFIG_FLASH_SIZE < STM32G4_FAMILY_MAX_FLASH)
+	/*
+	 * In case of bank1/2 discontinuity range, is invalid in both cases:
+	 *  - offest is in bank1, and end of range is beyond bank1
+	 *  - offset is between bank1 and bank2
+	 */
+	if (((offset < FLASH_SIZE / 2) && (offset + len > FLASH_SIZE / 2))
+	    || ((offset >= FLASH_SIZE / 2) && (offset < BANK2_OFFSET))) {
+		return 0;
+	}
```

Comment:
I think the number of comparisons could be reduced here while still getting the same result, but not sure if that makes it harder to read/understand. Up to you to decide.
```suggestion
	/*
	 * In case of bank1/2 discontinuity, the range should not end
	 * beyond bank1 and start before bank2 at the same time.
	 * Locations beyond bank2 are caught by flash_stm32_range_exists.
	 */
	if ((offset + len > FLASH_SIZE / 2) && (offset < BANK2_OFFSET)) {
		return 0;
	}
```

Reply:
Consider this a comment only and not a -1. Will approve after erwango's comments are addressed. And I will also give it another try in hardware in the next days.

Reply:
Done